spec/rubocop/cop/performance/redundant_split_regexp_argument_spec.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Performance::RedundantSplitRegexpArgument do
+  subject(:cop) { described_class.new }
+
+  it 'accepts methods other than split' do
+    expect_no_offenses("'a,b,c'.insert(2, 'a')")
+  end
+
+  it 'accepts when split does not receive a regexp' do
+    expect_no_offenses("'a,b,c'.split(',')")
+  end
+
+  it 'accepts when split does not receive a deterministic regexp' do
+    expect_no_offenses("'a,b,c'.split(/,+/)")
+    expect_no_offenses("'a,b,c'.split(/\./)")
+  end
+
+  it 'registers an offense when the method is split' do
+    expect_offense(<<~RUBY)
+      'a,b,c'.split(/,/)
+      ^^^^^^^^^^^^^^^^^^ Use string as argument instead of regexp.
```

Comment:
It will be more easy to understand for users if offense range is only argument :-)

```suggestion
      'a,b,c'.split(/,/)
                    ^^^ Use string as argument instead of regexp.
```

Reply:
How do I change that?

Reply:
This can be implemented by changing the argument of `add_offense`.

```diff
diff --git a/lib/rubocop/cop/performance/redundant_split_regexp_argument.rb b/lib/rubocop/cop/performance/redundant_split_regexp_argument.rb
index eacccf731..c7afad436 100644
--- a/lib/rubocop/cop/performance/redundant_split_regexp_argument.rb
+++ b/lib/rubocop/cop/performance/redundant_split_regexp_argument.rb
@@ -27,7 +27,7 @@ module RuboCop
           return unless split_call_with_regexp?(node)
           return unless determinist_regexp?(node.first_argument)

-          add_offense(node) do |corrector|
+          add_offense(node.first_argument) do |corrector|
             autocorrect(corrector, node)
           end
         end
```